.github/workflows/main.yml

```diff
@@ -1,5 +1,10 @@
 name: OctoBot-CI
-on: push
+on:
+  push:
+    branches:
+      - 'master'
+      - 'dev'
+  pull_request:
```

Comment:
is it fine if we keep it all in the same workflow ? I mean in the same .yml

Reply:
I don't understand your question, why spliting ?

Reply:
In the github doc example, they split workflows into 2 files (on with access to secrets and the other without), I'm wondering if this is required or not in our case (to protect secrets)

Reply:
They said :
> To do this in a secure manner, the untrusted code must be handled via the pull_request trigger so that it is isolated in an unprivileged environment

And the first example explains that `pull_request` give no access to secrets : 
```
# read-only repo token
# no access to secrets
on:
  pull_request:
```
The second and third example are completly different from the first one.

I agree one point : we'll need additional changes to enable our CI to succeed without secrets access (we currently have some unecessary secrets that should be replaced by hardcoded values).

Reply:
Right, what I'm wondering is if having both `push` and `pull_request` triggers on the same workflow is not making it so that `pull_request` get the `push` rights as well